src/foam/nanos/controller/ApplicationController.js

```diff
@@ -377,6 +378,19 @@ foam.CLASS({
       });
     },
 
+    function requestCapability(capabilityInfo) {
+      var self = this;
+      console.log("CAPABILITY SCREEN REQUESTED");
+      return new Promise(function(resolve, reject) {
+        self.stack.push({
+          class: 'foam.nanos.crunch.ui.CapabilityInterceptView',
+          data: self.__subContext__["capabilityDAO"],
+          capabilityOptions: capabilityInfo.capabilityOptions
+        });
+        self.loginSuccess$.sub(resolve);
```

Comment:
Huh, I guess I copied that there by accident but I'm not sure what I could replace it with. `capabilityAquired` would be ambiguous because there's no indication of what capability was acquired. It looks like loginSuccess is needed for some behaviours of ApplicationController though so it may be sufficient to just remove it here.

Reply:
> It looks like loginSuccess is needed for some behaviours of ApplicationController though so it may be sufficient to just remove it here.

I agree that `loginSuccess` isn't required for this feature, but we _do_ need something that serves a similar purpose for this feature. The point of `loginSuccess` is to block progress until the user logs in, then resume it. In the same way, we need to block progress until the user either gets a capability or decides they don't want to continue with whatever they were trying to do.

Reply:
Ah, the `.sub(resolve)` ties loginSuccess to the promise. My understanding now is that we could use just one property for capabilityAquired and reset it to false at the beginning of every requestCapability call. (this property would be a signal rather than meaningful information - is there a different way to do that?)

Reply:
> My understanding now is that we could use just one property for capabilityAquired and reset it to false at the beginning of every requestCapability call.

Yeah, that's right.

> this property would be a signal rather than meaningful information - is there a different way to do that?

I'd argue that the signal is meaningful.